### PR TITLE
Ming/get started page

### PR DESCRIPTION
**Pull request by mingbeam**
### Problems

- create lifestyle picture animation for banner
- create get started banner with animated background picture
- create a signup form for get started page
- create api to connect the signup form with mailchimp
- create get started page and implement all the created components

- edit copywriting and add non breaking space
- make the label move up when filling an input field or when the field is focused
- make success modal responsive

- link to get started page when click on 'Get Started' button on landing page
- make label on email and store name field working

### How would the reviewers verify the solutions

Screenshot (1537)

- make label move up
image
image

### Preflight checklist

- [x] **Manual testing** has been performed according to the acceptance criteria.
- [ ] _(heavy style change)_ The solution has been tested on the major browser (Chrome, Firefox, Safari, Microsoft Edge).

### CODE REVIEW

File: styles/getstarted.module.css [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+.spacer {
+  height: 80px;
+  margin-top: 80px;
+}
+
+.link {
+  font-weight: bold;
+  text-decoration: underline;
+}
```
Comment (oukunanan) [inline styles/getstarted.module.css:58]: Unused className?

File: styles/getstarted.module.css [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+.shapes_container {
+  position: absolute;
+  width: 110%;
+}
+
+.spacer {
+  height: 80px;
+  margin-top: 80px;
+}
```
Comment (oukunanan) [inline styles/getstarted.module.css:53]: Unuse className?

File: components/early_birds/GetStartedSignupForm.tsx [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+import * as Yup from 'yup'
+import styles from './GetStartedSignupForm.module.css'
+import axios from 'axios'
+import classNames from 'classnames'
+import SuccessModal from '../SuccessModal'
+import ErrorStateMessage from './ErrorStateMessage'
+import Text from '../Text'
+
+const phoneRegExp =
+  /^(([+\\]+[1-9]{1,4}[ \\-]*)|(\\([0-9]{2,3}\\)[ \\-]*)|([0-9]{2,4})[ \\-]*)*?[0-9]{3,4}?[ \\-]*[0-9]{3,4}?$/
```
Comment (oukunanan) [inline components/early_birds/GetStartedSignupForm.tsx:12]: We should avoid repetitive code as a [DRY principle](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)

Screen Shot 2565-01-31 at 11 07 36

File: components/early_birds/GetStartedSignupForm.tsx [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+            resetForm()
+            setErrorMessage('')
+          } catch (error: any) {
+            setErrorMessage(error.response.data.error)
+          }
+          setSubmitting(false)
+        }}
+      >
+        {(formik) => (
+          <Form className={styles.form_container}>
```
Comment (oukunanan) [inline components/early_birds/GetStartedSignupForm.tsx:58]: **Optional but I will give a task for refactoring for sure**: I think we can extract this into our own component.
Screen Shot 2565-01-31 at 11 12 00

File: components/early_birds/LifestylePicAnimation.module.css [hunk 5]
```diff
@@ -0,0 +1,3 @@
+.bg {
```
Comment (oukunanan) [inline components/early_birds/LifestylePicAnimation.module.css:1]: Better naming

File: components/early_birds/LifestylePicAnimation.tsx [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    config: { duration: 4000 },
+    onRest: (_a, _b, item) => {
+      if (index === item) {
+        set((state) => (state + 1) % slides.length)
+      }
+    },
+    exitBeforeEnter: true,
+  })
+  return (
+    <div>
```
Comment (oukunanan) [inline components/early_birds/LifestylePicAnimation.tsx:41]: How about a fragment?

File: components/early_birds/LifestylePicAnimation.tsx [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+  PictureThree,
+  PictureFour,
+  PictureFive,
+  PictureSix,
+  PictureSeven,
+  PictureEight,
+]
+
+export default function LifestylePicAnimation() {
+  const [index, set] = useState(0)
```
Comment (oukunanan) [inline components/early_birds/LifestylePicAnimation.tsx:26]:
```suggestion
  const [imageIndex, setimageIndex] = useState(0)
```

Comment (oukunanan) [reply components/early_birds/LifestylePicAnimation.tsx:26]: or slide

File: components/early_birds/GetStartedBanner.module.css [hunk 8]
```diff
@@ -0,0 +1,39 @@
+.container {
+  width: 100%;
+  height: 312px;
+  position: relative;
+  overflow: hidden;
```
[the remark this answers is not in the document]
Comment (mingbeam) [reply components/early_birds/GetStartedBanner.module.css]: removed

File: pages/getstarted.tsx [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+
+export default function GetStarted() {
+  const [mounted, setMounted] = useState(false)
+  useEffect(() => {
+    setMounted(true)
+  }, [])
+
+  const renderDescription = () => {
+    return (
+      <p className={styles.description}>
```
[the remark this answers is not in the document]
Comment (mingbeam) [reply pages/getstarted.tsx]: changed to h3

File: pages/getstarted.tsx [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+        <Text size={18} weight={600}>
+          Exclusive early access promo
+        </Text>
+        <Spacer height={4} />
+        <Text color="#535353" family="Prompt">
+          รับสิทธิพิเศษ สำหรับผู้ใช้งานกลุ่มแรก
+        </Text>
```
[the remark this answers is not in the document]
Comment (mingbeam) [reply pages/getstarted.tsx]: done

File: pages/getstarted.tsx [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+      <div className={styles.shapes_container}>
+        <Image src={backgroundShapes} alt="Background Shapes" />
+      </div>
+
+      <div className={styles.main_container}>
+        <GetStartedBanner />
+        {renderDescription()}
+        <SignupForm />
+
+        {mounted && (
```
[the remark this answers is not in the document]
Comment (mingbeam) [reply pages/getstarted.tsx]: yes, it works for this page but not for freeforthree page krub

File: styles/getstarted.module.css (hunk 1 above)
Comment (mingbeam) [reply styles/getstarted.module.css]: removed

File: styles/getstarted.module.css (hunk 2 above)
Comment (mingbeam) [reply styles/getstarted.module.css]: removed

File: components/early_birds/GetStartedSignupForm.tsx (hunk 3 above)
Comment (mingbeam) [reply components/early_birds/GetStartedSignupForm.tsx:12]: havent done this

File: components/early_birds/GetStartedSignupForm.tsx (hunk 4 above)
Comment (mingbeam) [reply components/early_birds/GetStartedSignupForm.tsx:58]: havent done this

File: components/early_birds/LifestylePicAnimation.module.css (hunk 5 above)
Comment (mingbeam) [reply components/early_birds/LifestylePicAnimation.module.css]: changed to image_container

File: components/early_birds/LifestylePicAnimation.tsx (hunk 6 above)
Comment (mingbeam) [reply components/early_birds/LifestylePicAnimation.tsx]: changed to a fragment

File: components/early_birds/LifestylePicAnimation.tsx (hunk 7 above)
Comment (mingbeam) [reply components/early_birds/LifestylePicAnimation.tsx]: changed to slideIndex and setSlideIndex

**Review by oukunanan — APPROVED** [review]
Approved. Please create a PR for refactoring. remove a lot of duplicate code and make it reusable.

**Comment by oukunanan** [comment]
Label click is not working on `Work Email` and `Store Name`

https://user-images.githubusercontent.com/86279170/152466643-6533d477-aeb2-43dd-9e7c-c1859e6f185c.mp4

**Review by oukunanan — CHANGES_REQUESTED** [review]
According to my latest comment.  👆🏼

**Review by oukunanan — APPROVED** [review]

**Pull request merged**